aep/general/0135/aep.md.j2

```diff
@@ -4,15 +4,45 @@ In REST APIs, it is customary to make a `DELETE` request to a resource's URI
 (for example, `/v1/publishers/{publisher}/books/{book}`) in order to delete
 that resource.
 
-Resource-oriented design (AIP-121) honors this pattern through the `Delete`
-method. These RPCs accept the URI representing that resource and usually return
+Resource-oriented design ([AEP-121][aep-121]) honors this pattern through the `Delete`
+method. These operations accept the URI representing that resource and usually return
 an empty response.
 
 ## Guidance
 
 APIs **should** generally provide a delete method for resources unless it is
 not valuable for users to do so.
 
+The Delete method **should** succeed if and only if a resource was present and
+was successfully deleted. If the resource did not exist, the method **should**
+send a `404 Not found` (`NOT_FOUND`) error.
+
+### Requests
+
+{% tab oas -%}
+
+Delete methods are specified using the following pattern:
+
+{% sample 'delete.oas.yaml', 'paths' %}
+
+- The HTTP verb **must** be `DELETE`.
+- There **must not** be a request body.
+  - If a delete request contains a body, the body **must** be ignored, and
+    **must not** cause an error.
+- The request **must not** require any fields in the query string. The request
+  **should not** include optional fields in the query string unless described
+  in another AEP.
+- Delete operations **must** return `204 No Content` with no response body.
+  - Exception: If the resource is soft deleted ([AIP-164][aep-164]), in which case the
```

Comment:
```suggestion
  - Exception: If the resource is soft deleted ([AIP-164][aep-164]), the
```

Reply:
But also, should this guidance be in the separate soft delete AEP?

Reply:
I argue this specific deviation should be removed.

We can support soft-deletes, but to add this clause that says *sometimes* you should return the resource always seemed like needless deviation. in practice no one uses that, and it makes making a resource soft-deletable a backwards incompatible change.

Reply:
The reason soft-deletes were referenced here was because API readers would rely on no resources being returned, and not realize that there is this edge case with soft deletes. Yet another reason to remove the deviation, IMO.

Reply:
I have removed the soft-delete return value edge case.

Reply:
We actually use soft delete at Roblox, and do return the resource in that case.

The way I think about it, using the soft delete pattern makes `Delete` into a specialized `Update`, and `Update` returns the resource.

Why would making a resource soft-deleteable be a breaking change?  If you previously got `{}` as a response and now get `{path: "..."  state: DELETED"}`, what code could that break?

Reply:
This is an example of the thing I mentioned in last week's meeting, where we end up debating changes to mature guidance from Google.  If there was some known improvement Google wanted to make but couldn't for consistency (e.g. `page_size` -> `max_page_size`) then we should definitely take the opportunity to make those changes for the AEPs...but is this one of those cases?

Reply:
> Why would making a resource soft-deleteable be a breaking change? If you previously got `{}` as a response and now get `{path: "..." state: DELETED"}`, what code could that break?

...I was thinking about JSON, not gRPC, whoops.  Changing response type from `Empty` to <resource> is obviously a breaking change for gRPC clients.

Reply:
After live discussion with Yusuke, I think I'm fine with these being removed.  I'll file a ticket for Yusuke to write a PR with his idea for an alternative design pattern for the use cases that might otherwise use soft delete.